### ISSUE TITLE
[Bug]: Kicking players for a too large payload can't be canceled

### ISSUE THREAD

**Issue by Timongcraft** [SpigotMC/BungeeCord#3509]
### Bungeecord version

git:BungeeCord-Bootstrap:1.20-R.02-SNAPSHOT:94ea027:1726 by md_5

### Server version

git-Paper-169 (MC: 1.20.1) (Implementing API version 1.20.1-R0.1-SNAPSHOT) (Git: b4e3b3d)

### Client version

1.20.1

### Bungeecord plugins

*none*

### The bug

If you send large packets you get a exception that "the payload is too large" and get kicked. Which in my opinion should just be managed by plugins and/or controlled by a config value.

### Log output (links)

```
07:28:04 [WARNING] [Timongcraft] -> UpstreamBridge - could not decode packet! java.lang.IllegalArgumentException: Payload too large
07:28:04 [INFO] [Timongcraft] disconnected with: DecoderException : java.lang.IllegalArgumentException: Payload too large @ io.netty.handler.codec.MessageToMessageDecoder:98
07:28:04 [INFO] [Timongcraft] -> UpstreamBridge has disconnected
```

### Checking

- [X] I am using BungeeCord and **not a fork**. Issues with forks should not be reported here.
- [X] I think this is **not** an issue with a bungeecord plugin.
- [ ] I have not read these checkboxes and therefore I just ticked them all.
- [X] This is not a question or plugin creation help request.

**Comment by Janmm14** [comment]
We are just enforcing vanilla protocol limits for the plugin message packet size.

Can you name any reasons on the reason you want this changed?

**Comment by Timongcraft** [comment]
E.g. the Axiom Mod (https://axiom.moulberry.com) with the Plugin (https://github.com/Moulberry/AxiomPaperPlugin) is a very useful building mod which sends large custom packets.

**Comment by Outfluencer** [comment]
They should fix it in my opinion, its not a bungee issue

**Comment by Timongcraft** [comment]
But why?
It works for example with velocity, so why should't it with bungeecord?

**Comment by Moulberry** [comment]
I agree that it's not a bungeecord issue, but it's not so easy to fix on my end.

The mod & plugin supports changing the packet size limit [here](https://github.com/Moulberry/AxiomPaperPlugin/blob/master/src/main/java/com/moulberry/axiom/AxiomPaper.java#L90) from 0x100000 to Short.MAX_VALUE, which would solve the disconnect.

However, doing so will result in extreme performance degradation due to the large number of packets and reduced compressability of a 32k buffer compared to a 1mb buffer.

As the original poster said, a reasonable middle-ground would be to introduce a config option to change these limits. If this is an acceptable solution, I'd be happy to submit a pull request to this project.

Thanks, Moulberry

**Comment by Janmm14** [comment]
I would prefer not making this a regular config option.

I suggest bungee doing one of these things:
- Extract constant to static field so plugins can edit easier(still without guarantees on api)
- Create hidden config option
- Use system property

**Comment by Ayoree** [comment]
If anyone still having this issue try to change `max-block-buffer-packet-size` in axiom settings to less than `0x100000`.
I changed this to `32767` and it works for me.

Here is original BungeeCord code, that causes this error:
https://github.com/SpigotMC/BungeeCord/blob/5e59b6dc8509fae7c482d4dc4dbda68f71e7b835/protocol/src/main/java/net/md_5/bungee/protocol/packet/PluginMessage.java#L65-L66
If you want to use bigger values in Axiom config you should recompile BungeeCord yourself

**Comment by Moulberry** [comment]
> If anyone still having this issue try to change `max-block-buffer-packet-size` in axiom settings to less than `0x100000`. I changed this to its minimal allowed value `32767` and it works for me.

The minimum value is 30000. (https://github.com/Moulberry/AxiomPaperPlugin/blob/09334064cc38ac0c8e420fd1277424873b6abe6e/src/main/resources/config.yml#L29)
I usually recommend using 31000, just to give a little bit of leeway.

**Comment by Ayoree** [comment]
> > If anyone still having this issue try to change `max-block-buffer-packet-size` in axiom settings to less than `0x100000`. I changed this to its minimal allowed value `32767` and it works for me.
> 
> The minimum value is 30000. (https://github.com/Moulberry/AxiomPaperPlugin/blob/09334064cc38ac0c8e420fd1277424873b6abe6e/src/main/resources/config.yml#L29) I usually recommend using 31000, just to give a little bit of leeway.

Thanks! I have updated my post
